be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -136,6 +136,22 @@ class SegmentIterator final : public ChunkIterator {
             bool may_has_del_row = chunk->delete_state() != DEL_NOT_SATISFIED;
             for (size_t i = 0; i < _column_iterators.size(); i++) {
                 const ColumnPtr& col = chunk->get_column_by_index(i);
+                if (config::enable_index_filter_column_prune && _prune_cols.count(i)) {
+                    // In general, we can skip the col resize operation here to save more memory.
+                    // But there are two cases resize is needed:
+                    // 1. It is the first column, in this case, column's size will be treated as
+                    //    the chunk size.
+                    // 2. In debug/ASAN mode, chunk::check_or_die will check all column have the
+                    //    same size or not.
+                    if (i == 0) {
```

Comment:
So tricky, better to resize always?

Reply:
This feels too tricky. It relies on num_rows() to obtain the row count of the first column. It is easy to cause problems later. If you take out the Column and then look at the row_count, there will be problems, and it is related to the basics data structures are coupled,

Reply:
Fixed it, resize all columns